src/commissioner/commissioner_api.h

```diff
@@ -0,0 +1,224 @@
+                                                                  bool             aAllowAll,
+                                                                  const uint8_t *  aJoinerEui64);
+
+/**
+ * This function frees steering data.
+ *
+ * @param[in]    aSteeringData          steering data to be freed
+ *
+ * @returns OTBR_COMMISSIONER_INVALID_ARGS if nullptr provided, OTBR_COMMISSIONER_SUCCESS otherwise
+ *
+ */
+otbr_commissioner_result_t otbr_commissioner_free_steering_data(steering_data_t aSteeringData);
+
+/**
+ * This function computes pskc for commissioner.
+ *
+ * @param[out]   aPskcBin           buffer to store computed pskc
+ * @param[in]    aPskcSize          size of aPskcBin
+ * @param[in]    aExtPanId          extended panid of thread network
+ * @param[in]    aNetworkName       thread network name
+ * @param[in]    aNetworkPassword   thread network password
+ *
+ * @returns OTBR_COMMISSIONER_INVALID_ARGS if nullptr provided, OTBR_COMMISSIONER_SUCCESS otherwise
+ *
+ */
+otbr_commissioner_result_t otbr_commissioner_compute_pskc(uint8_t *      aPskcBin,
+                                                          size_t         aPskcSize,
+                                                          const uint8_t *aExtPanId,
+                                                          const char *   aNetworkName,
+                                                          const char *   aNetworkPassword);
+
+/**
+ * This function create commissioner handle.
+ *
+ * @param[out]   aHandle            buffer to store commissioner handle
+ * @param[in]    aPskcBin           computed pskc
+ * @param[in]    aKeepAliveRate     rate of sending keep alive message to leader
+ *
+ * @returns OTBR_COMMISSIONER_INVALID_ARGS if nullptr provided, OTBR_COMMISSIONER_SUCCESS otherwise
+ *
+ */
+otbr_commissioner_result_t otbr_commissioner_create_commissioner_handle(otbr_commissioner_handle_t *aHandle,
+                                                                        const uint8_t *             aPskcBin,
+                                                                        int                         aKeepAliveRate);
+
+/**
+ * This function free commissioner handle.
+ *
+ * @param[in]    aHandle            commissioner handle to be freed
+ *
+ * @returns OTBR_COMMISSIONER_INVALID_ARGS if nullptr provided, OTBR_COMMISSIONER_SUCCESS otherwise
+ *
+ */
+otbr_commissioner_result_t otbr_commissioner_free_commissioner_handle(otbr_commissioner_handle_t aHandle);
+
+/**
+ * This function set joiner pskd and steering data.
+ *
+ * @param[in]    aHandle            commissioner handle
+ * @param[in]    aPskdAscii         joiner pskd in ascii format
+ * @param[in]    aSteeringData      steering data to filter joiner
+ *
+ * @returns OTBR_COMMISSIONER_INVALID_ARGS if nullptr provided, OTBR_COMMISSIONER_SUCCESS otherwise
+ *
+ */
+otbr_commissioner_result_t otbr_commissioner_set_joiner(otbr_commissioner_handle_t aHandle,
```

Comment:
Thread Commissioner allows multiple joiners simultaneously. This API seems to limit to one joiner. Have we considered adding support for multiple joiners?

Reply:
The implementation itself does not limit number of joiners since it just sends JOINER.SET coap message, will change api name.

Reply:
This is limited by mJoinerSession dtls session. Will add support for multiple joiners afterwards.